src/uploaders/ReactNativeUploader.ts

```diff
@@ -66,11 +69,39 @@ class ReactNativeUploader {
 
   private async getSourceMapAndBundle(
     projectRoot: string,
-    retrieval: SourceMapRetrieval
+    retrieval: SourceMapRetrieval,
+    platform: Platform,
+    dev: boolean
   ): Promise<SourceMapBundlePair> {
     switch (retrieval.type) {
-      case SourceMapRetrievalType.Fetch:
-        throw new Error('SourceMapRetrievalType.Fetch is not implemented!')
+      case SourceMapRetrievalType.Fetch: {
+        const queryString = qs.stringify({ platform, dev })
+        const sourceMapUrl = `${retrieval.url}/index.js.map?${queryString}`
+        const bundleUrl = `${retrieval.url}/index.bundle?${queryString}`
+        let sourceMap
+        let bundle
+
+        try {
+          this.logger.debug(`Fetching source map from ${sourceMapUrl}`)
+          sourceMap = await fetch(sourceMapUrl)
+        } catch (e) {
+          this.logger.error(`Unable to fetch source map from ${retrieval.url}. Is the server running?`)
+          throw e
+        }
+
+        try {
+          this.logger.debug(`Fetching bundle from ${bundleUrl}`)
+          bundle = await fetch(bundleUrl)
+        } catch (e) {
+          this.logger.error(`Unable to fetch bundle from ${retrieval.url}. Is the server running?`)
```

Comment:
In this case the error message doesn't really make sense — if we got a source map then the server must be running. I'm not sure there's anything useful we can log in this case, so maybe just remove the second sentence?

Reply:
I mean the server could died between the two requests, so it's possible that it would make sense? You might be able to tell from the error whether it was trying to talk to a server that wasn't up, with the code `ECONNREFUSED` or something? Though maybe this is overthinking things.

I guess it's more likely for the failure reason to be something else. Happy for the message to be removed.